src/main/java/mate/academy/internetshop/dao/jdbc/UserDaoJdbcImpl.java

```diff
@@ -0,0 +1,199 @@
+package mate.academy.internetshop.dao.jdbc;
+
+import java.sql.Connection;
+import java.sql.PreparedStatement;
+import java.sql.ResultSet;
+import java.sql.SQLException;
+import java.sql.Statement;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Optional;
+import mate.academy.internetshop.dao.BucketDao;
+import mate.academy.internetshop.dao.UserDao;
+import mate.academy.internetshop.exceptions.AuthenticationException;
+import mate.academy.internetshop.lib.Dao;
+import mate.academy.internetshop.lib.Inject;
+import mate.academy.internetshop.model.Bucket;
+import mate.academy.internetshop.model.User;
+import org.apache.log4j.Logger;
+
+@Dao
+public class UserDaoJdbcImpl extends AbstractDao<User> implements UserDao {
+    private static Logger logger = Logger.getLogger(UserDaoJdbcImpl.class);
+    private static String DB_NAME = "internetshop";
+
+    @Inject
+    private static BucketDao bucketDao;
+
+    public UserDaoJdbcImpl(Connection connection) {
+        super(connection);
+    }
+
+    @Override
+    public User create(User user) {
+        String query = "INSERT INTO " + DB_NAME
+                + ".users (name, surname, login, password, token) VALUES (?, ?, ?, ?, ?);";
+        try (PreparedStatement statement
+                     = connection.prepareStatement(query, Statement.RETURN_GENERATED_KEYS)) {
+            statement.setString(1, user.getName());
+            statement.setString(2, user.getSurname());
+            statement.setString(3, user.getLogin());
+            statement.setString(4, user.getPassword());
+            statement.setString(5, user.getToken());
+            statement.executeUpdate();
+            try (ResultSet generatedKeys = statement.getGeneratedKeys()) {
+                if (generatedKeys.next()) {
+                    user.setId(generatedKeys.getLong(1));
+                } else {
+                    throw new SQLException("Add user failed, no ID obtained.");
+                }
+                bucketDao.create(new Bucket(user.getId()));
+                addRole(user);
+            }
+        } catch (SQLException e) {
+            logger.error("Can't create user", e);
+            return null;
+        }
+        return user;
+    }
+
+    @Override
+    public void addRole(User user) {
```

Comment:
```suggestion
    public void addRole(User user, Role role) {
```
@gkhrshch Якщо ти додаєш роль - то краще явно вказати яку роль ти додаєш